src/balance.rs

```diff
@@ -26,41 +41,219 @@ impl<T: Config> From<Error<T>> for StatusCode {
+        if self.balances.borrow().contains_key(account)
+            || self.initial_state.borrow().contains_key(account)
+        {
+            return Err(DispatchError::Other("initial balance already inserted"));
+        }
+        // TODO (neutrinoks): Enable this feature to ensure, that balance is available!
+        // self.ensure_can_withdraw(account, balance)?;
+
+        {
+            let mut balances = self.balances.borrow_mut();
+            balances.insert(account.clone(), *balance);
+        }
+        {
+            let mut initial_state = self.initial_state.borrow_mut();
+            initial_state.insert(account.clone(), *balance);
+        }
+
+        Ok(())
+    }
+
+    /// Executes the true transactions on the blockchain/substrate side after execution of
+    /// Move-script.
+    pub fn apply_transactions(&self) -> DispatchResult {
+        use core::cmp::Ordering;
+
+        self.cmp_with_initial_state()?;
+
+        let balances = self.balances.borrow();
+        let mut purse = NegativeImbalanceOf::<T>::zero();
+        let mut depts = Vec::<(&AccountIdOf<T>, BalanceOf<T>)>::new();
+
+        // Calculate balance differences and withdraw negative ones from user's accounts.
+        for (account, balance) in balances.iter() {
+            let initial_state = self.initial_state.borrow();
+            let true_balance = initial_state.get(account).unwrap();
+            match (*true_balance).cmp(balance) {
+                Ordering::Greater => {
+                    let dept = *true_balance - *balance;
+                    let imbalance = T::Currency::withdraw(
+                        account,
+                        dept,
+                        WithdrawReasons::TRANSFER,
+                        ExistenceRequirement::AllowDeath,
+                    )?;
+                    purse = purse.merge(imbalance);
+                }
+                Ordering::Less => {
+                    let dept = *balance - *true_balance;
+                    depts.push((account, dept));
+                }
+                Ordering::Equal => {}
+            }
+        }
+
+        // Now deposit depts from purse to new owners.
+        for (account, balance) in depts.into_iter() {
+            let imbalance = purse.extract(balance);
+            T::Currency::resolve_creating(account, imbalance);
+        }
+
+        Ok(())
+    }
+
+    /// Ensures that user can withdraw that given amount of money, which eventually will be used
+    /// within the Move-script execution.
+    #[allow(dead_code)]
+    fn ensure_can_withdraw(
+        &self,
+        account: &AccountIdOf<T>,
+        amount: &BalanceOf<T>,
+    ) -> DispatchResult {
+        if *amount < T::Currency::free_balance(account) {
+            Ok(())
+        } else {
+            Err(Error::<T>::InsufficientBalance.into())
+        }
+    }
+
+    /// Method ensures that a certain user exists in our balances record.
+    fn ensure_account_exists(&self, user: &AccountIdOf<T>) {
```

Comment:
More proper name would be 'create_account_in_the_chequelist'
`ensure_account_exists` sounds like it's gonna fail everything in case it doesn't exist.

Either way, we don't really need this function - we can create these _account entries_ with `cheques.entry(from).or_insert(0);` within the transfer function directly.